File: tests/testdata/codegen/unknown-resource-pp/nodejs/unknown-resource.ts
import * as pulumi from "@pulumi/pulumi";
import * as unknown from "@pulumi/unknown";

const provider = new pulumi.providers.Unknown("provider", {});
const main = new unknown.index.Main("main", {
    first: "hello",
    second: {
        foo: "bar",
    },
});
const fromModule: unknown.eks.Example[] = [];
for (let range = 0; range < 10; range++) {
    fromModule.push(new unknown.eks.Example(`fromModule-${range}`, {associatedMain: main.id}));
}
export const mainId = main.id;
export const values = fromModule.values.first;

File: tests/testdata/codegen/unknown-resource-pp/python/unknown-resource.py
import pulumi
from typing import Any
import pulumi_unknown as unknown

provider = pulumi.providers.Unknown("provider")
main = unknown.Main("main",
    first=hello,
    second={
        foo: bar,
    })
from_module: list[unknown.eks.Example] = []
for from_module_range in [{"value": i} for i in range(0, 10)]:
    from_module.append(unknown.eks.Example(f"fromModule-{from_module_range['value']}", associated_main=main.id))
pulumi.export("mainId", main["id"])
pulumi.export("values", from_module["values"]["first"])
